third_party/move/tools/move-unit-test/src/test_runner.rs

```diff
@@ -283,7 +291,7 @@ impl SharedTestingConfig {
         }
 
         let test_run_info = TestRunInfo::new(function_name.to_string(), now.elapsed());
-        match session.finish_with_extensions() {
+        match session.finish_with_extensions(&module_storage) {
```

Comment:
What else if not "full storage"? In theory, we only need metadata, so if we split module storage so that it consists of multiple traits: why not. What do you think?